lib/datadog/tracing.rb

```diff
@@ -0,0 +1,207 @@
+require 'ddtrace/configuration/validation_proxy'
+
+module Datadog
+  # Datadog APM tracing public API.
+  #
+  # The Datadog team ensures that public methods in this module
+  # only receive backwards compatible changes, and breaking changes
+  # will only occur in new major versions releases.
+  # @public_api
+  module Tracing
+    class << self
+      # (see Datadog::Tracer#trace)
+      # @public_api
+      def trace(name, continue_from: nil, **span_options, &block)
+        tracer.trace(name, continue_from: continue_from, **span_options, &block)
+      end
+
+      # (see Datadog::Tracer#continue_trace!)
+      # @public_api
+      def continue_trace!(digest, &block)
+        tracer.continue_trace!(digest, &block)
+      end
+
+      # The currently active {Datadog::Tracer} instance.
+      #
+      # The instance returned can change throughout the lifetime of the application.
+      # This means it is not advisable to cache it.
+      #
+      # The tracer can be configured through {.configure},
+      # through {Datadog::Configuration::Settings::DSL::Tracer} options.
+      #
+      # @deprecated Use public API methods available in {Datadog::Tracing} instead.
+      # @return [Datadog::Tracer] the active tracer
+      # @!attribute [r] tracer
+      # @public_api
+      def tracer
+        components.tracer
+      end
+
+      # The tracer's internal logger instance.
+      # All tracing log output is handled by this object.
+      #
+      # The logger can be configured through {.configure},
+      # through {Datadog::Configuration::Settings::DSL::Logger} options.
+      #
+      # @!attribute [r] logger
+      # @public_api
+      def logger
+        Datadog.logger
+      end
+
+      # Current tracer configuration.
+      #
+      # To modify the configuration, use {.configure}.
+      #
+      # @return [Datadog::Configuration::Settings]
+      # @!attribute [r] configuration
+      # @public_api
+      def configuration
+        Datadog.configuration
+      end
+
+      # Apply configuration changes to `ddtrace`. An example of a {.configure} call:
+      # ```
+      # Datadog::Tracing.configure do |c|
+      #   c.sampling.default_rate = 1.0
+      #   c.use :aws
+      #   c.use :rails
+      #   c.use :sidekiq
+      #   # c.diagnostics.debug = true # Enables debug output
+      # end
+      # ```
+      #
+      # Because many configuration changes require restarting internal components,
+      # invoking {.configure} is the only safe way to change `ddtrace` configuration.
```

Comment:
Tiny fix:

```suggestion
      # Apply configuration changes to `Datadog::Tracing`. An example of a {.configure} call:
      # ```
      # Datadog::Tracing.configure do |c|
      #   c.sampling.default_rate = 1.0
      #   c.use :aws
      #   c.use :rails
      #   c.use :sidekiq
      #   # c.diagnostics.debug = true # Enables debug output
      # end
      # ```
      #
      # Because many configuration changes require restarting internal components,
      # invoking {.configure} is the only safe way to change `Datadog::Tracing` configuration.
```

(The intent is to clarify that this applies to the tracing feature, not ddtrace-the-gem)